gui/elements/qt/_dimensions.py

```diff
@@ -69,17 +75,17 @@ def update_slider(self, axis, max_axis_length):
             # slider.setTickInterval(tick_interval)
             slider.setSingleStep(1)
 
-            def value_changed():
-                self.viewer.dimensions.indices[axis] = slider.value()
-                self.viewer.dimensions._need_redraw = True
-                self.viewer.dimensions._update()
-
-            slider.valueChanged.connect(value_changed)
-
             grid.addWidget(slider, row, 0)
             self.sliders.append(slider)
         else:
             slider = slider.widget()
 
+        slider.valueChanged.connect(lambda value, axis=axis: self._slider_value_changed(value, axis))
         slider.setMaximum(max_axis_length - 1)
+        self.setFixedHeight((dims-2)*19)
```

Comment:
Should make this "magic number" a constant

Reply:
have made it a constant